### ISSUE TITLE
Bad splitting of long lines

### ISSUE THREAD

**Issue by micheleCTDE** [uncrustify/uncrustify#3887]
Uncrustify 6d21e020

Input file
```
void main()
{
	snprintf(&ret[strlen(ret)], ret_max-strlen(ret), "viewBox=\"%lf %lf %lf %lf\"\n", icon->x, icon->y, icon->width, icon->height);
}
```

config file: see attached zip.

output file
```
void main()
{
	snprintf(&ret[strlen(ret)], ret_max - strlen(
						ret), "viewBox=\"%lf %lf %lf %lf\"\n", icon->x, icon->y, icon->width, icon->height);
}
```

The long line is splitted very awkwardly. This is caused by the wrong open parenthesis detection in the code.

Working on a fix, will upload PR when ready

[bad_split.zip](https://github.com/uncrustify/uncrustify/files/10030679/bad_split.zip)

**Comment by micheleCTDE** [comment]
This is not an easy one, there are lot of different cases. PR will take a while I guess.

**Comment by micheleCTDE** [comment]
Prior to the resolution of this issue, issue #3105 needs to be solved.

**Comment by micheleCTDE** [comment]
Solved by PR #4117.
